src/sonic-eventd/rsyslog_plugin/main.cpp

```diff
@@ -0,0 +1,49 @@
+#include <iostream>
+#include <unistd.h>
+#include "rsyslog_plugin.h"
+#include "syslog_parser.h"
+
+using namespace std;
+
+void showUsage() {
+    cerr << "Usage for rsyslog_plugin: " << " <option(s)> SOURCES\n"
+        << "Options:\n"
+        << "\t-r,required,type=string\t\tPath to regex file"
+        << "\t-m,required,type=string\t\tYANG module name of source generating syslog message"
+        << endl;
+}
+
+int main(int argc, char** argv) {
+    string regex_path;
+    string module_name;
+    int option_val;
+
+    while((option_val = getopt(argc, argv, "r:m:")) != -1) {
+        switch(option_val) {
+            case 'r':
+                if(optarg != NULL) {
+                    regex_path = optarg;
+                }
+                break;
+            case 'm':
+                if(optarg != NULL) {
+                    module_name = optarg;
+                }
+                break;
+            default:
+                showUsage();
+                return 1;
+        }
+    }
+
+    if(regex_path.empty() || module_name.empty()) { // Missing required rc path
+        showUsage();
+        return 1;
+    }
+    
+    SyslogParser* parser = new SyslogParser({}, json::array());
+    RsyslogPlugin* plugin = new RsyslogPlugin(parser, module_name, regex_path);
+
+    plugin->run();
```

Comment:
Curious:
If this plugin crashes or exits however, will rsyslog reload this plugin?

Reply:
From what I've seen while testing, it does reload every time a log message comes for a process specified in the .conf file even if there was an issue earlier.

Reply:
Please confirm:
In good scenario/run by plugin where it does not crash, rsyslog.d loads it just *once* only. Correct?

Reply:
Yes, loads only once